Client/Frontend/Home/ActivityStreamSupplementary.swift

```diff
@@ -0,0 +1,310 @@
+        titleLabel.backgroundColor = UIColor.whiteColor()
+        imageView.backgroundColor = UIColor.whiteColor()
+        contentView.backgroundColor = UIColor.whiteColor()
+    }
+
+}
+
+class ASHorizontalScrollCell: UITableViewCell {
+    private var collectionView: UICollectionView!
+    private var pageControl: UIPageControl!
+    private var headerView: ASHeaderView!
+
+    var headerTitle: String = "" {
+        didSet {
+            headerView.title = headerTitle
+        }
+    }
+
+    override init(style: UITableViewCellStyle, reuseIdentifier: String?) {
+        super.init(style: style, reuseIdentifier: reuseIdentifier)
+
+        let layout  = UICollectionViewFlowLayout()
+        layout.scrollDirection = UICollectionViewScrollDirection.Horizontal
+        layout.minimumLineSpacing = 0
+        layout.sectionInset = UIEdgeInsets(top: 0, left: 5, bottom: 0, right: 5)
+        backgroundColor = UIColor(white: 1.0, alpha: 0.5)
+
+        collectionView = UICollectionView(frame: CGRectZero, collectionViewLayout: layout)
+        collectionView.registerClass(TopSiteCell.self, forCellWithReuseIdentifier: "TopSiteCell")
+        collectionView.backgroundColor = UIColor(white: 1.0, alpha: 0.5)
+        collectionView.showsHorizontalScrollIndicator = false
+        collectionView.pagingEnabled = true
+
+        contentView.addSubview(collectionView)
+        collectionView.snp_makeConstraints { make in
+            make.edges.equalTo(contentView).offset(UIEdgeInsetsMake(18, 0, 0, 0))
+            make.height.equalTo(100)
+        }
+
+        //Page control will need to be swapped out with a thirdparty one. I cant customize the built in one at all
+        pageControl = UIPageControl()
+        pageControl.pageIndicatorTintColor = UIColor.grayColor()
+        pageControl.currentPageIndicatorTintColor = UIColor.darkGrayColor()
+        contentView.addSubview(pageControl)
+        pageControl.snp_makeConstraints { make in
+            make.width.equalTo(30)
+            make.height.equalTo(20)
+            make.top.equalTo(contentView)
+            make.trailing.equalTo(self.snp_trailing).offset(-5)
+        }
+
+        headerView = ASHeaderView(frame: CGRect.zero)
+        contentView.addSubview(headerView)
+        headerView.snp_makeConstraints { make in
+            make.width.equalTo(self.snp_width)
+            make.top.equalTo(self.snp_top)
+            make.bottom.equalTo(collectionView.snp_top)
+        }
+    }
+
+    func numberOfPagesChanged(sections: Int) {
+        pageControl.numberOfPages = sections
+    }
+
+    func currentPageChanged(currentPage: Int) {
+        pageControl.currentPage = currentPage
+    }
+
+    func setDelegate(delegate: ASHorizontalScrollSource) {
+        collectionView.delegate = delegate
+        collectionView.dataSource = delegate
+        delegate.numberOfPagesChangedHandler = numberOfPagesChanged
+        delegate.pageChangedHandler = currentPageChanged
```

Comment:
I created function handlers to help observe changes to the collectionview which can update pagecontrol. This way I dont expose pageControl outside of its class. I wish there were properties I could observe directly on collectionview without having to make the cell a delegate. But couldnt find any. Thoughts on this approach compared to just exposing pagecontrol?